runtime/cudaq/builder/kernel_builder.h

```diff
@@ -444,24 +453,27 @@ class kernel_builder : public details::kernel_builder_base {
     QuakeValue v(*opBuilder, param);                                           \
     details::NAME(*opBuilder, v, empty, qubit);                                \
   }                                                                            \
-  template <NumericType ConstantParameter>                                     \
-  void NAME(ConstantParameter &&constant, QuakeValue qubit) {                  \
-    NAME(constant, qubit);                                                     \
-  }                                                                            \
-  template <typename mod,                                                      \
+  template <typename mod, QuakeValueOrNumericType paramT,                      \
             typename =                                                         \
                 typename std::enable_if_t<std::is_same_v<mod, cudaq::adj>>>    \
-  void NAME(QuakeValue parameter, QuakeValue qubit) {                          \
-    NAME(-parameter, qubit);                                                   \
+  void NAME(paramT &&parameter, QuakeValue qubit) {                            \
+    if constexpr (NumericType<paramT>)                                         \
+      NAME(QuakeValue(*opBuilder, -parameter), qubit);                         \
+    else                                                                       \
+      NAME(-parameter, qubit);                                                 \
   }                                                                            \
-  template <typename mod, typename... QubitValues,                             \
+  template <typename mod, QuakeValueOrNumericType paramT,                      \
+            typename... QubitValues,                                           \
             typename = typename std::enable_if_t<sizeof...(QubitValues) >= 2>> \
-  void NAME(QuakeValue parameter, QubitValues... args) {                       \
+  void NAME(paramT &&parameter, QubitValues... args) {                         \
```

Comment:
Should we just go ahead and update this to `const ParamT&` and the one above it?

Reply:
It's a good idea. Changed both to `const ParamT&`.